/* $DragonFly: src/usr.bin/vi/patches/ex_write.c.patch,v 1.1 2006/06/24 00:04:18 sephe Exp $ */
--- ex_write.c	2006-06-23 23:08:03.000000000 +0800
+++ ex_write.c	2006-06-23 23:08:54.000000000 +0800
@@ -352,14 +352,17 @@
 	    S_ISREG(sb.st_mode) && fsync(fileno(fp)))
 		goto err;
 
-	if (fclose(fp))
+	if (fclose(fp)) {
+		fp = NULL;
 		goto err;
+	}
 
 	rval = 0;
 	if (0) {
 err:		if (!F_ISSET(sp->ep, F_MULTILOCK))
 			msgq_str(sp, M_SYSERR, name, "%s");
-		(void)fclose(fp);
+		if (fp != NULL)
+			fclose(fp);
 		rval = 1;
 	}
 
